src/main/kotlin/org/amshove/kluent/Collections.kt

```diff
@@ -206,51 +206,51 @@ infix fun Short.shouldBeIn(theArray: ShortArray) = this.apply { this shouldBeIn
+infix fun <T, I : Iterable<T>> I.shouldContainNone(things: Iterable<T>) = this.apply { assertTrue("Expected $this to contain none of $things", this.none { things.contains(it) }) }
 
-infix fun <T> Iterable<T>.shouldContainNone(things: Array<T>) = this.apply { assertTrue("Expected $this to contain none of $things", this.none { things.contains(it) }) }
+infix fun <T, I : Iterable<T>> I.shouldContainNone(things: Array<T>) = this.apply { assertTrue("Expected $this to contain none of $things", this.none { things.contains(it) }) }
 
-infix fun <T> Iterable<T>.shouldContainAll(things: Iterable<T>) = this.apply { things.forEach { shouldContain(it) } }
+infix fun <T, I : Iterable<T>> I.shouldContainAll(things: Iterable<T>) = this.apply { things.forEach { shouldContain(it) } }
 
-infix fun <T> Iterable<T>.shouldContainAll(things: Array<T>) = this.apply { things.forEach { shouldContain(it) } }
+infix fun <T, I : Iterable<T>> I.shouldContainAll(things: Array<T>) = this.apply { things.forEach { shouldContain(it) } }
 
-infix fun <T> Iterable<T>.shouldNotContain(theThing: T) = this.apply { if (!this.contains(theThing)) Unit else fail("$this should not contain $theThing", "the Iterable to not contain $theThing", join(this)) }
+infix fun <T, I : Iterable<T>> I.shouldNotContain(theThing: T) = this.apply { if (!this.contains(theThing)) Unit else fail("$this should not contain $theThing", "the Iterable to not contain $theThing", join(this)) }
 
-infix fun <T> Iterable<T>.shouldNotContainAny(things: Iterable<T>) = this.apply { things.forEach { shouldNotContain(it) } }
+infix fun <T, I : Iterable<T>> I.shouldNotContainAny(things: Iterable<T>) = this.apply { things.forEach { shouldNotContain(it) } }
 
-infix fun <T> Iterable<T>.shouldNotContainAny(things: Array<T>) = this.apply { things.forEach { shouldNotContain(it) } }
+infix fun <T, I : Iterable<T>> I.shouldNotContainAny(things: Array<T>) = this.apply { things.forEach { shouldNotContain(it) } }
 
-infix fun <T> Iterable<T>?.shouldEqual(expected: Iterable<T>?) = this.apply { assertEquals(expected, this) }
+infix fun <T, I: Iterable<T>> I.shouldEqual(expected: Iterable<T>?) = this.apply { assertEquals(expected, this) }
 
-fun <T> Iterable<T>.shouldBeEmpty() = this.apply { assertEmpty(this, "Iterable") }
+fun <I : Iterable<*>> I.shouldBeEmpty() = this.apply { assertEmpty(this, "Iterable") }
 
-fun <T> Iterable<T>.shouldNotBeEmpty() = this.apply { assertNotEmpty(this, "Iterable") }
+fun <I : Iterable<*>> I.shouldNotBeEmpty() = this.apply { assertNotEmpty(this, "Iterable") }
 
-infix fun <R, T> Map<R, T>.shouldHaveKey(theKey: R) = this.apply { if (this.containsKey(theKey)) Unit else fail("$this should contain key $theKey", "$theKey", join(this.keys)) }
+infix fun <K, V, M : Map<K, V>> M.shouldHaveKey(theKey: K) = this.apply { if (this.containsKey(theKey)) Unit else fail("$this should contain key $theKey", "$theKey", join(this.keys)) }
```

Comment:
Do we need the generic `V` here?

Reply:
Where checking for the key or the value either one is needed (I'll drop the other one)